client/app/scripts/superdesk/directives/sdUserAvatar.js

```diff
@@ -5,27 +5,37 @@ define(['angular'], function (angular) {
     return angular.module('superdesk.avatar', [])
         .directive('sdUserAvatar', function() {
             return {
-                scope: {src: '=', initials: '='},
+                scope: {user: '='},
                 link: function (scope, element, attrs) {
-
-                    var figure = element.parents('figure');
+                    scope.$watch('user', function (user) {
+                        if (user) {
+                            initAvatar(user);
+                        }
+                    });
 
                     element.on('error', function (e) {
```

Comment:
now you can handle it gracefully and display initials also when it can't fetch user image that should be there..